src/main/dynamo/Beyonce.ts

```diff
@@ -280,7 +280,37 @@ export class Beyonce {
       Put: { TableName: this.table.tableName, Item: item }
     }))
 
-    await this.client.transactWrite({ TransactItems: encryptedItems }).promise()
+    await this.client
+      .transactWrite({ TransactItems: encryptedItems })
+      .promise()
+  }
+
+  /** Write multiple items into Dynamo.
+   */
+  async batchPut<T extends TaggedModel>(params: {
+    items: T[];
+  }): Promise<{ unprocessedItems: T[] }> {
+    const maybeEncryptedItems = await this.maybeEncryptItems(params.items)
+    const encryptedItems = maybeEncryptedItems.map(item => ({
+      PutRequest: {
+        Item: item
+      }
+    }))
+
+    const { UnprocessedItems } = await this.client
+      .batchWrite({
+        RequestItems: {
+          [this.table.tableName]: encryptedItems
+        }
+      })
+      .promise()
+    let unprocessedItems: T[] = []
+    if (UnprocessedItems && UnprocessedItems[this.table.tableName]) {
+      unprocessedItems = UnprocessedItems[this.table.tableName]
```

Comment:
How about something like this:

```TypeScript
const unprocessedItems: T[] = []

if (UnprocessedItems && UnprocessedItems[tableName]) {
  UnprocessedItems[tableName].forEach((request) => {
      if (request.PutRequest && request.PutRequest.Item) {
         const json = toJSON<T>(request.PutRequest.Item)
         unprocessedItems.push(json)
      }
  }) 
}

return { unprocessedItems }
```

It: 

1. Avoids two passes through the list
2. Avoids asserting a potentially undefined field is defined (i.e. avoids the `!`)